src/styles/main.scss

```diff
@@ -0,0 +1,217 @@
+* {
+  box-sizing: border-box;
+}
+
+html {
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+}
+
+body {
+  margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+.calendar {
+  gap: 1px;
+  display: flex;
+  flex-wrap: wrap;
+  width: 707px;
+  margin: auto;
+}
+
+.calendar:nth-child(7n) {
+  margin-top: 0;
+  padding-left: 1px;
+}
+
+.calendar .calendar__day {
+  width: 100px;
+  height: 100px;
+  background-color: #eee;
+  border: 1px solid #000;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  text-align: center;
+  transition: all 0.5s;
+  cursor: pointer;
+  padding-left: 1px;
+}
+
+.calendar .calendar__day:nth-child(1)::before {
+  content: '1';
+}
+
+.calendar .calendar__day:nth-child(2)::before {
+  content: '2';
+}
```

Comment:
You need to add number of days with @for 
https://sass-lang.com/documentation/at-rules/control/for/